src/plugins/discover/public/application/main/hooks/use_text_based_query_language.ts

```diff
@@ -120,14 +121,16 @@ export function useTextBasedQueryLanguage({
               prev.current.query = query[language];
               prev.current.columns = nextColumns;
             }
-            const nextState = {
-              ...(addDataViewToState && { index: dataViewObj.id }),
-              ...((addColumnsToState || queryChanged) && { columns: nextColumns }),
-              ...(viewMode === VIEW_MODE.AGGREGATED_LEVEL && {
-                viewMode: getValidViewMode({ viewMode, isTextBasedQueryMode: true }),
-              }),
-            };
-            await stateContainer.appState.replaceUrlState(nextState);
+            // just change URL state if necessary
+            if (addDataViewToState || addColumnsToState || changeViewMode) {
+              const nextState = {
+                ...(addDataViewToState && { index: undefined }),
+                ...(addColumnsToState && { columns: nextColumns }),
+                ...(changeViewMode && { viewMode: undefined }),
```

Comment:
Are we setting `index` and `viewMode` to `undefined` to clear them from the URL since they're invalid params for ES|QL?

Reply:
will follow up here tomorrow, thx for all your feedback, I'll address

yes, the idea is to clean up state that's not relevant in ES|QL ... we could keep it there, because it doesn't hurt. But I think removing params that are not relevant, is a way to declutter state, and in this case the url.